cmd/agent/app/config.go

```diff
@@ -212,14 +216,24 @@ func (b *Builder) CreateAgent(mFactory metrics.Factory, logger *zap.Logger) (*Ag
 		b.CollectorServiceName = defaultCollectorServiceName
 	}
 
+	// Combine CollectorHostPort and CollectorHostPorts to
+	// form a FixedDiscoverer if applicable.
+	var hostPorts []string
+	hostPorts = append(hostPorts, b.CollectorHostPorts...)
+	if b.CollectorHostPort != "" {
+		hostPorts = strings.Split(b.CollectorHostPort, ",")
+	}
+	if len(hostPorts) != 0 {
+		d := discovery.FixedDiscoverer(hostPorts)
+		notifier := &discovery.Dispatcher{}
+		b.WithDiscoverer(d).WithDiscoveryNotifier(notifier)
+	}
```

Comment:
added. During adding unittest, I realize there is potential partially overlapping case for the two sources of static collector lists, and the handling potentially be very complicated, for example, localhost:5700 vs 127.0.0.1:5700. I start to wonder that why do not we just use CollectotHostPort string, and get rid of CollectorHostPorts entirely. By this way, the logic will be a lot simpler.